modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/streaming/PagingProviderProducer.java

```diff
@@ -83,6 +88,7 @@ public PagingProviderProducer(PagingProvider<Object, T> delegate,
     this.supportsOAuth = supportsOAuth;
     retryPolicy = (RetryPolicyTemplate) executionContext.getRetryPolicyTemplate().orElseGet(NoRetryPolicyTemplate::new);
     connectionSupplierFactory = createConnectionSupplierFactory();
+    tryToMutateConfigurationStats(config, (MutableConfigurationStats::addActiveComponent));
```

Comment:
this `tryToMutateConfigurationStats` is overkill. Just extract the the stats object in the constructor and make sure it's mutable there. No need to extract it, test it and instantiate a lamba every time